storage/blob.go

```diff
@@ -801,6 +802,49 @@ func (b BlobStorageClient) GetPageRanges(container, name string) (GetPageRangesR
 	return out, err
 }
 
+// PutAppendBlob initializes an empty append blob with specified name. An
+// append blob must be created using this method before appending blocks.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179451.aspx
+func (b BlobStorageClient) PutAppendBlob(container, name string, extraHeaders map[string]string) error {
+	path := fmt.Sprintf("%s/%s", container, name)
+	uri := b.client.getEndpoint(blobServiceName, path, url.Values{})
+	headers := b.client.getStandardHeaders()
+	headers["x-ms-blob-type"] = string(BlobTypeAppend)
+	headers["Content-Length"] = "0"
```

Comment:
@BrianBland Hmm, I don't think you will still need this header now that you updated `buildCanonicalizedString` with

```diff
+	contentLength := headers["Content-Length"]
+	if contentLength == "0" {
+		contentLength = ""
+	}
```

In fact, there should be more occurrences of `headers["Content-Length"] = "0"` in this file, I think we can remove all those if you don't mind doing this in PR as well?